Hi @fjuma Thank you for welcoming my contribution. As instructed, I have split up my commit into 7 commits in which each tackles an ELY issue. Please let me know if any other change is required from my end, I'll be happy to do them.

Also, I want to mention that I haven't been able to successfully run `mvn clean install` on my local just yet. I'm not sure if these are due to my changes or if it's due to me not having configured MVN settings/configuration correctly. In README it's mentioned to [setup JBOSS MVN repository](https://github.com/wildfly-security/wildfly-elytron/tree/2.x#setup-the-jboss-maven-repository) using setting.xml, but I didn't quite understand how to do that. I am guessing that it might be due to it.


Here is the error output I'm getting:
```
-------------------------------------------------------
 T E S T S
-------------------------------------------------------
Running org.wildfly.security.ssl.CipherSuiteSelectorTest
Tests run: 32, Failures: 0, Errors: 32, Skipped: 0, Time elapsed: 0.783 sec <<< FAILURE! - in org.wildfly.security.ssl.CipherSuiteSelectorTest
testSingleSuiteUsingOpensslName(org.wildfly.security.ssl.CipherSuiteSelectorTest)  Time elapsed: 0.492 sec  <<< ERROR!
java.lang.ExceptionInInitializerError
        at org.wildfly.security.ssl.CipherSuiteSelector.evaluate(CipherSuiteSelector.java:286)
        at org.wildfly.security.ssl.CipherSuiteSelectorTest.getSelectedSuites(CipherSuiteSelectorTest.java:53)
        at org.wildfly.security.ssl.CipherSuiteSelectorTest.testSingleSuiteUsingOpensslName(CipherSuiteSelectorTest.java:103)
Caused by: java.lang.IllegalArgumentException: Logger implementation class org.wildfly.security.ssl.ElytronMessages_$logger could not be instantiated
        at org.jboss.logging.Logger.doGetMessageLogger(Logger.java:2579)
        at org.jboss.logging.Logger.getMessageLogger(Logger.java:2530)
        at org.jboss.logging.Logger.getMessageLogger(Logger.java:2516)
        at org.wildfly.security.ssl.ElytronMessages.<clinit>(ElytronMessages.java:56)
        ... 3 more
Caused by: java.lang.Error: Unresolved compilation problems: 
        The type javax.annotation.Generated is not accessible
        Generated cannot be resolved to a type

        at org.wildfly.security.ssl.ElytronMessages_$logger.<init>(ElytronMessages_$logger.java:6)
        at org.jboss.logging.Logger.doGetMessageLogger(Logger.java:2573)
        ... 6 more

testNotRsaAfterRsa(org.wildfly.security.ssl.CipherSuiteSelectorTest)  Time elapsed: 0.004 sec  <<< ERROR!
java.lang.NoClassDefFoundError: Could not initialize class org.wildfly.security.ssl.ElytronMessages
        at org.wildfly.security.ssl.CipherSuiteSelector.evaluate(CipherSuiteSelector.java:286)
        at org.wildfly.security.ssl.CipherSuiteSelectorTest.getSelectedSuites(CipherSuiteSelectorTest.java:53)
        at org.wildfly.security.ssl.CipherSuiteSelectorTest.testNotRsaAfterRsa(CipherSuiteSelectorTest.java:205)

testInvalidFormatCipherSuites(org.wildfly.security.ssl.CipherSuiteSelectorTest)  Time elapsed: 0.003 sec  <<< ERROR!
java.lang.NoClassDefFoundError: Could not initialize class org.wildfly.security.ssl.ElytronMessages
        at org.wildfly.security.ssl.CipherSuiteSelector.fromNamesString(CipherSuiteSelector.java:516)
        at org.wildfly.security.ssl.CipherSuiteSelectorTest.testInvalidFormatCipherSuites(CipherSuiteSelectorTest.java:310)

testMinusAesBetweenRsaAndAes(org.wildfly.security.ssl.CipherSuiteSelectorTest)  Time elapsed: 0 sec  <<< ERROR!
java.lang.NoClassDefFoundError: Could not initialize class org.wildfly.security.ssl.ElytronMessages
        at org.wildfly.security.ssl.CipherSuiteSelector.evaluate(CipherSuiteSelector.java:286)
        at org.wildfly.security.ssl.CipherSuiteSelectorTest.getSelectedSuites(CipherSuiteSelectorTest.java:53)
        at org.wildfly.security.ssl.CipherSuiteSelectorTest.testMinusAesBetweenRsaAndAes(CipherSuiteSelectorTest.java:184)

testStrengthForAllAndComplementofall(org.wildfly.security.ssl.CipherSuiteSelectorTest)  Time elapsed: 0 sec  <<< ERROR!
java.lang.NoClassDefFoundError: Could not initialize class org.wildfly.security.ssl.ElytronMessages
        at org.wildfly.security.ssl.CipherSuiteSelector.evaluate(CipherSuiteSelector.java:286)
        at org.wildfly.security.ssl.CipherSuiteSelectorTest.getSelectedSuites(CipherSuiteSelectorTest.java:53)
        at org.wildfly.security.ssl.CipherSuiteSelectorTest.testStrengthForAllAndComplementofall(CipherSuiteSelectorTest.java:227)

testAll(org.wildfly.security.ssl.CipherSuiteSelectorTest)  Time elapsed: 0.003 sec  <<< ERROR!
java.lang.NoClassDefFoundError: Could not initialize class org.wildfly.security.ssl.ElytronMessages
        at org.wildfly.security.ssl.CipherSuiteSelector.evaluate(CipherSuiteSelector.java:286)
        at org.wildfly.security.ssl.CipherSuiteSelectorTest.getSelectedSuites(CipherSuiteSelectorTest.java:53)
        at org.wildfly.security.ssl.CipherSuiteSelectorTest.testAll(CipherSuiteSelectorTest.java:58)

testComplementofdefault(org.wildfly.security.ssl.CipherSuiteSelectorTest)  Time elapsed: 0.001 sec  <<< ERROR!
java.lang.NoClassDefFoundError: Could not initialize class org.wildfly.security.ssl.ElytronMessages
        at org.wildfly.security.ssl.CipherSuiteSelector.evaluate(CipherSuiteSelector.java:286)
        at org.wildfly.security.ssl.CipherSuiteSelectorTest.getSelectedSuites(CipherSuiteSelectorTest.java:53)
        at org.wildfly.security.ssl.CipherSuiteSelectorTest.testComplementofdefault(CipherSuiteSelectorTest.java:87)

testSeparatorColon(org.wildfly.security.ssl.CipherSuiteSelectorTest)  Time elapsed: 0.008 sec  <<< ERROR!
java.lang.NoClassDefFoundError: Could not initialize class org.wildfly.security.ssl.ElytronMessages
        at org.wildfly.security.ssl.CipherSuiteSelector.evaluate(CipherSuiteSelector.java:286)
        at org.wildfly.security.ssl.CipherSuiteSelectorTest.getSelectedSuites(CipherSuiteSelectorTest.java:53)
        at org.wildfly.security.ssl.CipherSuiteSelectorTest.testSeparatorColon(CipherSuiteSelectorTest.java:265)

testSeparatorComma(org.wildfly.security.ssl.CipherSuiteSelectorTest)  Time elapsed: 0 sec  <<< ERROR!
java.lang.NoClassDefFoundError: Could not initialize class org.wildfly.security.ssl.ElytronMessages
        at org.wildfly.security.ssl.CipherSuiteSelector.evaluate(CipherSuiteSelector.java:286)
        at org.wildfly.security.ssl.CipherSuiteSelectorTest.getSelectedSuites(CipherSuiteSelectorTest.java:53)
        at org.wildfly.security.ssl.CipherSuiteSelectorTest.testSeparatorComma(CipherSuiteSelectorTest.java:257)

testSeparatorSpace(org.wildfly.security.ssl.CipherSuiteSelectorTest)  Time elapsed: 0 sec  <<< ERROR!
java.lang.NoClassDefFoundError: Could not initialize class org.wildfly.security.ssl.ElytronMessages
        at org.wildfly.security.ssl.CipherSuiteSelector.evaluate(CipherSuiteSelector.java:286)
        at org.wildfly.security.ssl.CipherSuiteSelectorTest.getSelectedSuites(CipherSuiteSelectorTest.java:53)
        at org.wildfly.security.ssl.CipherSuiteSelectorTest.testSeparatorSpace(CipherSuiteSelectorTest.java:249)

testSingleSuiteUsingStandardName(org.wildfly.security.ssl.CipherSuiteSelectorTest)  Time elapsed: 0 sec  <<< ERROR!
java.lang.NoClassDefFoundError: Could not initialize class org.wildfly.security.ssl.ElytronMessages
        at org.wildfly.security.ssl.CipherSuiteSelector.evaluate(CipherSuiteSelector.java:286)
        at org.wildfly.security.ssl.CipherSuiteSelectorTest.getSelectedSuites(CipherSuiteSelectorTest.java:53)
        at org.wildfly.security.ssl.CipherSuiteSelectorTest.testSingleSuiteUsingStandardName(CipherSuiteSelectorTest.java:95)

testDoublePlusBetweenAnonAndTls12(org.wildfly.security.ssl.CipherSuiteSelectorTest)  Time elapsed: 0.001 sec  <<< ERROR!
java.lang.NoClassDefFoundError: Could not initialize class org.wildfly.security.ssl.ElytronMessages
        at org.wildfly.security.ssl.CipherSuiteSelector.evaluate(CipherSuiteSelector.java:286)
        at org.wildfly.security.ssl.CipherSuiteSelectorTest.getSelectedSuites(CipherSuiteSelectorTest.java:53)
        at org.wildfly.security.ssl.CipherSuiteSelectorTest.testDoublePlusBetweenAnonAndTls12(CipherSuiteSelectorTest.java:135)

testMinusBetweenRsaAndAes(org.wildfly.security.ssl.CipherSuiteSelectorTest)  Time elapsed: 0.001 sec  <<< ERROR!
java.lang.NoClassDefFoundError: Could not initialize class org.wildfly.security.ssl.ElytronMessages
        at org.wildfly.security.ssl.CipherSuiteSelector.evaluate(CipherSuiteSelector.java:286)
        at org.wildfly.security.ssl.CipherSuiteSelectorTest.getSelectedSuites(CipherSuiteSelectorTest.java:53)
        at org.wildfly.security.ssl.CipherSuiteSelectorTest.testMinusBetweenRsaAndAes(CipherSuiteSelectorTest.java:176)

testValidCipherSuites(org.wildfly.security.ssl.CipherSuiteSelectorTest)  Time elapsed: 0.02 sec  <<< ERROR!
java.lang.NoClassDefFoundError: Could not initialize class org.wildfly.security.ssl.ElytronMessages
        at org.wildfly.security.ssl.CipherSuiteSelector.evaluate(CipherSuiteSelector.java:286)
        at org.wildfly.security.ssl.CipherSuiteSelectorTest.getSelectedSuites(CipherSuiteSelectorTest.java:51)
        at org.wildfly.security.ssl.CipherSuiteSelectorTest.testValidCipherSuites(CipherSuiteSelectorTest.java:292)

testMinusBetweenRsaAes(org.wildfly.security.ssl.CipherSuiteSelectorTest)  Time elapsed: 0.004 sec  <<< ERROR!
java.lang.Exception: Unexpected exception, expected<java.lang.IllegalArgumentException> but was<java.lang.NoClassDefFoundError>
        at org.junit.internal.runners.statements.ExpectException.evaluate(ExpectException.java:30)
        at org.junit.runners.ParentRunner$3.evaluate(ParentRunner.java:306)
        at org.junit.runners.BlockJUnit4ClassRunner$1.evaluate(BlockJUnit4ClassRunner.java:100)
        at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:366)
        at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:103)
        at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:63)
        at org.junit.runners.ParentRunner$4.run(ParentRunner.java:331)
        at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:79)
        at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:329)
        at org.junit.runners.ParentRunner.access$100(ParentRunner.java:66)
        at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:293)
        at org.junit.runners.ParentRunner$3.evaluate(ParentRunner.java:306)
        at org.junit.runners.ParentRunner.run(ParentRunner.java:413)
        at org.apache.maven.surefire.junit4.JUnit4Provider.execute(JUnit4Provider.java:367)
        at org.apache.maven.surefire.junit4.JUnit4Provider.executeWithRerun(JUnit4Provider.java:274)
        at org.apache.maven.surefire.junit4.JUnit4Provider.executeTestSet(JUnit4Provider.java:238)
        at org.apache.maven.surefire.junit4.JUnit4Provider.invoke(JUnit4Provider.java:161)
        at org.apache.maven.surefire.booter.ForkedBooter.invokeProviderInSameClassLoader(ForkedBooter.java:290)
        at org.apache.maven.surefire.booter.ForkedBooter.runSuitesInProcess(ForkedBooter.java:242)
        at org.apache.maven.surefire.booter.ForkedBooter.main(ForkedBooter.java:121)
Caused by: java.lang.NoClassDefFoundError: Could not initialize class org.wildfly.security.ssl.ElytronMessages
        at org.wildfly.security.ssl.CipherSuiteSelector.fromString(CipherSuiteSelector.java:484)
        at org.wildfly.security.ssl.CipherSuiteSelectorTest.testMinusBetweenRsaAes(CipherSuiteSelectorTest.java:200)

testNotRsaBetweenRsaAndRsa(org.wildfly.security.ssl.CipherSuiteSelectorTest)  Time elapsed: 0 sec  <<< ERROR!
java.lang.NoClassDefFoundError: Could not initialize class org.wildfly.security.ssl.ElytronMessages
        at org.wildfly.security.ssl.CipherSuiteSelector.evaluate(CipherSuiteSelector.java:286)
        at org.wildfly.security.ssl.CipherSuiteSelectorTest.getSelectedSuites(CipherSuiteSelectorTest.java:53)
        at org.wildfly.security.ssl.CipherSuiteSelectorTest.testNotRsaBetweenRsaAndRsa(CipherSuiteSelectorTest.java:212)

testCombinedDefaultCipherSuites(org.wildfly.security.ssl.CipherSuiteSelectorTest)  Time elapsed: 0.001 sec  <<< ERROR!
java.lang.NoClassDefFoundError: Could not initialize class org.wildfly.security.ssl.ElytronMessages
        at org.wildfly.security.ssl.CipherSuiteSelector.evaluate(CipherSuiteSelector.java:286)
        at org.wildfly.security.ssl.CipherSuiteSelector$3.evaluate(CipherSuiteSelector.java:556)
        at org.wildfly.security.ssl.CipherSuiteSelectorTest.testCombinedDefaultCipherSuites(CipherSuiteSelectorTest.java:319)

testMinusAesAfterRsa(org.wildfly.security.ssl.CipherSuiteSelectorTest)  Time elapsed: 0.005 sec  <<< ERROR!
java.lang.NoClassDefFoundError: Could not initialize class org.wildfly.security.ssl.ElytronMessages
        at org.wildfly.security.ssl.CipherSuiteSelector.evaluate(CipherSuiteSelector.java:286)
        at org.wildfly.security.ssl.CipherSuiteSelectorTest.getSelectedSuites(CipherSuiteSelectorTest.java:53)
        at org.wildfly.security.ssl.CipherSuiteSelectorTest.testMinusAesAfterRsa(CipherSuiteSelectorTest.java:168)

testPlusBeforeFirstRsa(org.wildfly.security.ssl.CipherSuiteSelectorTest)  Time elapsed: 0 sec  <<< ERROR!
java.lang.NoClassDefFoundError: Could not initialize class org.wildfly.security.ssl.ElytronMessages
        at org.wildfly.security.ssl.CipherSuiteSelector.evaluate(CipherSuiteSelector.java:286)
        at org.wildfly.security.ssl.CipherSuiteSelectorTest.getSelectedSuites(CipherSuiteSelectorTest.java:53)
        at org.wildfly.security.ssl.CipherSuiteSelectorTest.testPlusBeforeFirstRsa(CipherSuiteSelectorTest.java:159)

testSingleTlsSuiteUsingSslPrefixName(org.wildfly.security.ssl.CipherSuiteSelectorTest)  Time elapsed: 0.003 sec  <<< ERROR!
java.lang.NoClassDefFoundError: Could not initialize class org.wildfly.security.ssl.ElytronMessages
        at org.wildfly.security.ssl.CipherSuiteSelector.evaluate(CipherSuiteSelector.java:286)
        at org.wildfly.security.ssl.CipherSuiteSelectorTest.getSelectedSuites(CipherSuiteSelectorTest.java:53)
        at org.wildfly.security.ssl.CipherSuiteSelectorTest.testSingleTlsSuiteUsingSslPrefixName(CipherSuiteSelectorTest.java:111)

testPlusBetweenAnonAndTls12AndAfter(org.wildfly.security.ssl.CipherSuiteSelectorTest)  Time elapsed: 0.005 sec  <<< ERROR!
java.lang.NoClassDefFoundError: Could not initialize class org.wildfly.security.ssl.ElytronMessages
        at org.wildfly.security.ssl.CipherSuiteSelector.evaluate(CipherSuiteSelector.java:286)
        at org.wildfly.security.ssl.CipherSuiteSelectorTest.getSelectedSuites(CipherSuiteSelectorTest.java:53)
        at org.wildfly.security.ssl.CipherSuiteSelectorTest.testPlusBetweenAnonAndTls12AndAfter(CipherSuiteSelectorTest.java:143)

testSeparatorCommaSpace(org.wildfly.security.ssl.CipherSuiteSelectorTest)  Time elapsed: 0 sec  <<< ERROR!
java.lang.NoClassDefFoundError: Could not initialize class org.wildfly.security.ssl.ElytronMessages
        at org.wildfly.security.ssl.CipherSuiteSelector.evaluate(CipherSuiteSelector.java:286)
        at org.wildfly.security.ssl.CipherSuiteSelectorTest.getSelectedSuites(CipherSuiteSelectorTest.java:53)
        at org.wildfly.security.ssl.CipherSuiteSelectorTest.testSeparatorCommaSpace(CipherSuiteSelectorTest.java:273)

testDefault(org.wildfly.security.ssl.CipherSuiteSelectorTest)  Time elapsed: 0 sec  <<< ERROR!
java.lang.NoClassDefFoundError: Could not initialize class org.wildfly.security.ssl.ElytronMessages
        at org.wildfly.security.ssl.CipherSuiteSelector.evaluate(CipherSuiteSelector.java:286)
        at org.wildfly.security.ssl.CipherSuiteSelectorTest.getSelectedSuites(CipherSuiteSelectorTest.java:53)
        at org.wildfly.security.ssl.CipherSuiteSelectorTest.testDefault(CipherSuiteSelectorTest.java:79)

testMinusRsaBetweenRsaAndRsa(org.wildfly.security.ssl.CipherSuiteSelectorTest)  Time elapsed: 0 sec  <<< ERROR!
java.lang.NoClassDefFoundError: Could not initialize class org.wildfly.security.ssl.ElytronMessages
        at org.wildfly.security.ssl.CipherSuiteSelector.evaluate(CipherSuiteSelector.java:286)
        at org.wildfly.security.ssl.CipherSuiteSelectorTest.getSelectedSuites(CipherSuiteSelectorTest.java:53)
        at org.wildfly.security.ssl.CipherSuiteSelectorTest.testMinusRsaBetweenRsaAndRsa(CipherSuiteSelectorTest.java:192)

testTLS12NameInCipherSuites(org.wildfly.security.ssl.CipherSuiteSelectorTest)  Time elapsed: 0 sec  <<< ERROR!
java.lang.NoClassDefFoundError: Could not initialize class org.wildfly.security.ssl.ElytronMessages
        at org.wildfly.security.ssl.CipherSuiteSelector.fromNamesString(CipherSuiteSelector.java:516)
        at org.wildfly.security.ssl.CipherSuiteSelectorTest.testTLS12NameInCipherSuites(CipherSuiteSelectorTest.java:301)

testMultiplePlus(org.wildfly.security.ssl.CipherSuiteSelectorTest)  Time elapsed: 0 sec  <<< ERROR!
java.lang.NoClassDefFoundError: Could not initialize class org.wildfly.security.ssl.ElytronMessages
        at org.wildfly.security.ssl.CipherSuiteSelector.evaluate(CipherSuiteSelector.java:286)
        at org.wildfly.security.ssl.CipherSuiteSelectorTest.getSelectedSuites(CipherSuiteSelectorTest.java:53)
        at org.wildfly.security.ssl.CipherSuiteSelectorTest.testMultiplePlus(CipherSuiteSelectorTest.java:151)

testPlusBetweenAnonAndNullEncryption(org.wildfly.security.ssl.CipherSuiteSelectorTest)  Time elapsed: 0.001 sec  <<< ERROR!
java.lang.NoClassDefFoundError: Could not initialize class org.wildfly.security.ssl.ElytronMessages
        at org.wildfly.security.ssl.CipherSuiteSelector.evaluate(CipherSuiteSelector.java:286)
        at org.wildfly.security.ssl.CipherSuiteSelectorTest.getSelectedSuites(CipherSuiteSelectorTest.java:53)
        at org.wildfly.security.ssl.CipherSuiteSelectorTest.testPlusBetweenAnonAndNullEncryption(CipherSuiteSelectorTest.java:119)

testStrengthForComplementofallAndAll(org.wildfly.security.ssl.CipherSuiteSelectorTest)  Time elapsed: 0.001 sec  <<< ERROR!
java.lang.NoClassDefFoundError: Could not initialize class org.wildfly.security.ssl.ElytronMessages
        at org.wildfly.security.ssl.CipherSuiteSelector.evaluate(CipherSuiteSelector.java:286)
        at org.wildfly.security.ssl.CipherSuiteSelectorTest.getSelectedSuites(CipherSuiteSelectorTest.java:53)
        at org.wildfly.security.ssl.CipherSuiteSelectorTest.testStrengthForComplementofallAndAll(CipherSuiteSelectorTest.java:238)

testComplementofall(org.wildfly.security.ssl.CipherSuiteSelectorTest)  Time elapsed: 0 sec  <<< ERROR!
java.lang.NoClassDefFoundError: Could not initialize class org.wildfly.security.ssl.ElytronMessages
        at org.wildfly.security.ssl.CipherSuiteSelector.evaluate(CipherSuiteSelector.java:286)
        at org.wildfly.security.ssl.CipherSuiteSelectorTest.getSelectedSuites(CipherSuiteSelectorTest.java:53)
        at org.wildfly.security.ssl.CipherSuiteSelectorTest.testComplementofall(CipherSuiteSelectorTest.java:71)

testPlusBetweenAnonAndTls12(org.wildfly.security.ssl.CipherSuiteSelectorTest)  Time elapsed: 0 sec  <<< ERROR!
java.lang.NoClassDefFoundError: Could not initialize class org.wildfly.security.ssl.ElytronMessages
        at org.wildfly.security.ssl.CipherSuiteSelector.evaluate(CipherSuiteSelector.java:286)
        at org.wildfly.security.ssl.CipherSuiteSelectorTest.getSelectedSuites(CipherSuiteSelectorTest.java:53)
        at org.wildfly.security.ssl.CipherSuiteSelectorTest.testPlusBetweenAnonAndTls12(CipherSuiteSelectorTest.java:127)

testNotAesBetweenRsaAndAes(org.wildfly.security.ssl.CipherSuiteSelectorTest)  Time elapsed: 0 sec  <<< ERROR!
java.lang.NoClassDefFoundError: Could not initialize class org.wildfly.security.ssl.ElytronMessages
        at org.wildfly.security.ssl.CipherSuiteSelector.evaluate(CipherSuiteSelector.java:286)
        at org.wildfly.security.ssl.CipherSuiteSelectorTest.getSelectedSuites(CipherSuiteSelectorTest.java:53)
        at org.wildfly.security.ssl.CipherSuiteSelectorTest.testNotAesBetweenRsaAndAes(CipherSuiteSelectorTest.java:219)

testTLS13NameInCipherList(org.wildfly.security.ssl.CipherSuiteSelectorTest)  Time elapsed: 0 sec  <<< ERROR!
java.lang.NoClassDefFoundError: Could not initialize class org.wildfly.security.ssl.ElytronMessages
        at org.wildfly.security.ssl.CipherSuiteSelector.fromString(CipherSuiteSelector.java:484)
        at org.wildfly.security.ssl.CipherSuiteSelectorTest.testTLS13NameInCipherList(CipherSuiteSelectorTest.java:282)

Running org.wildfly.security.ssl.MechanismDatabaseTest
Tests run: 3, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.717 sec - in org.wildfly.security.ssl.MechanismDatabaseTest

Results :

Tests in error: 
  CipherSuiteSelectorTest.testAll:58->getSelectedSuites:53 » NoClassDefFound Cou...
  CipherSuiteSelectorTest.testCombinedDefaultCipherSuites:319 » NoClassDefFound ...
  CipherSuiteSelectorTest.testComplementofall:71->getSelectedSuites:53 » NoClassDefFound
  CipherSuiteSelectorTest.testComplementofdefault:87->getSelectedSuites:53 » NoClassDefFound
  CipherSuiteSelectorTest.testDefault:79->getSelectedSuites:53 » NoClassDefFound
  CipherSuiteSelectorTest.testDoublePlusBetweenAnonAndTls12:135->getSelectedSuites:53 » NoClassDefFound
  CipherSuiteSelectorTest.testInvalidFormatCipherSuites:310 » NoClassDefFound Co...
  CipherSuiteSelectorTest.testMinusAesAfterRsa:168->getSelectedSuites:53 » NoClassDefFound
  CipherSuiteSelectorTest.testMinusAesBetweenRsaAndAes:184->getSelectedSuites:53 » NoClassDefFound
  CipherSuiteSelectorTest.testMinusBetweenRsaAes »  Unexpected exception, expect...
  CipherSuiteSelectorTest.testMinusBetweenRsaAndAes:176->getSelectedSuites:53 » NoClassDefFound
  CipherSuiteSelectorTest.testMinusRsaBetweenRsaAndRsa:192->getSelectedSuites:53 » NoClassDefFound
  CipherSuiteSelectorTest.testMultiplePlus:151->getSelectedSuites:53 » NoClassDefFound
  CipherSuiteSelectorTest.testNotAesBetweenRsaAndAes:219->getSelectedSuites:53 » NoClassDefFound
  CipherSuiteSelectorTest.testNotRsaAfterRsa:205->getSelectedSuites:53 » NoClassDefFound
  CipherSuiteSelectorTest.testNotRsaBetweenRsaAndRsa:212->getSelectedSuites:53 » NoClassDefFound
  CipherSuiteSelectorTest.testPlusBeforeFirstRsa:159->getSelectedSuites:53 » NoClassDefFound
  CipherSuiteSelectorTest.testPlusBetweenAnonAndNullEncryption:119->getSelectedSuites:53 » NoClassDefFound
  CipherSuiteSelectorTest.testPlusBetweenAnonAndTls12:127->getSelectedSuites:53 » NoClassDefFound
  CipherSuiteSelectorTest.testPlusBetweenAnonAndTls12AndAfter:143->getSelectedSuites:53 » NoClassDefFound
  CipherSuiteSelectorTest.testSeparatorColon:265->getSelectedSuites:53 » NoClassDefFound
  CipherSuiteSelectorTest.testSeparatorComma:257->getSelectedSuites:53 » NoClassDefFound
  CipherSuiteSelectorTest.testSeparatorCommaSpace:273->getSelectedSuites:53 » NoClassDefFound
  CipherSuiteSelectorTest.testSeparatorSpace:249->getSelectedSuites:53 » NoClassDefFound
  CipherSuiteSelectorTest.testSingleSuiteUsingOpensslName:103->getSelectedSuites:53 » ExceptionInInitializer
  CipherSuiteSelectorTest.testSingleSuiteUsingStandardName:95->getSelectedSuites:53 » NoClassDefFound
  CipherSuiteSelectorTest.testSingleTlsSuiteUsingSslPrefixName:111->getSelectedSuites:53 » NoClassDefFound
  CipherSuiteSelectorTest.testStrengthForAllAndComplementofall:227->getSelectedSuites:53 » NoClassDefFound
  CipherSuiteSelectorTest.testStrengthForComplementofallAndAll:238->getSelectedSuites:53 » NoClassDefFound
  CipherSuiteSelectorTest.testTLS12NameInCipherSuites:301 » NoClassDefFound Coul...
  CipherSuiteSelectorTest.testTLS13NameInCipherList:282 » NoClassDefFound Could ...
  CipherSuiteSelectorTest.testValidCipherSuites:292->getSelectedSuites:51 » NoClassDefFound

Tests run: 35, Failures: 0, Errors: 32, Skipped: 0

[INFO] ------------------------------------------------------------------------
[INFO] Reactor Summary for WildFly Elytron Parent 2.3.0.CR1-SNAPSHOT:
[INFO] 
[INFO] WildFly Elytron Parent ............................. SUCCESS [  4.129 s]
[INFO] WildFly Elytron - ASN.1 ............................ SUCCESS [  8.891 s]
[INFO] WildFly Elytron - Base ............................. SUCCESS [  2.288 s]
[INFO] WildFly Elytron - Util ............................. SUCCESS [  5.822 s]
[INFO] WildFly Elytron - Auth ............................. SUCCESS [  0.508 s]
[INFO] WildFly Elytron - Provider Util .................... SUCCESS [  0.829 s]
[INFO] WildFly Elytron - X.500 ............................ SUCCESS [  3.531 s]
[INFO] WildFly Elytron - X.500 Certificate Utility Classes  SUCCESS [  0.661 s]
[INFO] WildFly Elytron - X.500 Certificates ............... SUCCESS [  7.065 s]
[INFO] WildFly Elytron - Credential ....................... SUCCESS [ 11.617 s]
[INFO] WildFly Elytron - Digest ........................... SUCCESS [  0.421 s]
[INFO] WildFly Elytron - Credential ....................... SUCCESS [  3.943 s]
[INFO] WildFly Elytron - Permission ....................... SUCCESS [  1.955 s]
[INFO] WildFly Elytron - Auth Server ...................... SUCCESS [  7.485 s]
[INFO] WildFly Elytron - Audit Logging .................... SUCCESS [  0.661 s]
[INFO] WildFly Elytron - Encryption ....................... SUCCESS [  3.833 s]
[INFO] WildFly Elytron - Password Implementation .......... SUCCESS [ 25.503 s]
[INFO] WildFly Elytron - Security Manager Action .......... SUCCESS [  0.371 s]
[INFO] WildFly Elytron - Mechanism GSSAPI ................. SUCCESS [  0.610 s]
[INFO] WildFly Elytron - Auth Util ........................ SUCCESS [  5.288 s]
[INFO] WildFly Elytron - X.500 Principals ................. SUCCESS [  0.379 s]
[INFO] WildFly Elytron - Realm ............................ SUCCESS [ 11.849 s]
[INFO] WildFly Elytron - Realm JDBC ....................... SUCCESS [  7.466 s]
[INFO] WildFly Elytron - Credential Store ................. SUCCESS [  7.291 s]
[INFO] WildFly Elytron - Credential Source Implementation . SUCCESS [  0.600 s]
[INFO] WildFly Elytron - HTTP ............................. SUCCESS [  0.717 s]
[INFO] WildFly Elytron - Mechanism ........................ SUCCESS [  0.717 s]
[INFO] WildFly Elytron - SSL .............................. FAILURE [  7.160 s]
[INFO] WildFly Elytron - SASL ............................. SKIPPED
[INFO] WildFly Elytron - SSH Util ......................... SKIPPED
[INFO] WildFly Elytron - Client ........................... SKIPPED
[INFO] WildFly Elytron - Realm LDAP ....................... SKIPPED
[INFO] WildFly Elytron - JSON Utility Classes ............. SKIPPED
[INFO] WildFly Elytron - Realm Token ...................... SKIPPED
[INFO] WildFly Elytron - Auth Server HTTP ................. SKIPPED
[INFO] WildFly Elytron - Auth Server SASL ................. SKIPPED
[INFO] WildFly Elytron - Auth Server Deprecated ........... SKIPPED
[INFO] WildFly Elytron - Credential Source Deprecated ..... SKIPPED
[INFO] WildFly Elytron - Mechanism Http ................... SKIPPED
[INFO] WildFly Elytron - HTTP Basic ....................... SKIPPED
[INFO] WildFly Elytron - HTTP Bearer ...................... SKIPPED
[INFO] WildFly Elytron - HTTP Cert ........................ SKIPPED
[INFO] WildFly Elytron - Mechanism Digest ................. SKIPPED
[INFO] WildFly Elytron - HTTP DIGEST ...................... SKIPPED
[INFO] WildFly Elytron - HTTP External .................... SKIPPED
[INFO] WildFly Elytron - HTTP Basic ....................... SKIPPED
[INFO] WildFly Elytron - HTTP SPNEGO ...................... SKIPPED
[INFO] WildFly Elytron - HTTP Util ........................ SKIPPED
[INFO] WildFly Elytron - HTTP ............................. SKIPPED
[INFO] WildFly Elytron - Jose Util ........................ SKIPPED
[INFO] WildFly Elytron - JWK .............................. SKIPPED
[INFO] WildFly Elytron - HTTP Stateful Basic .............. SKIPPED
[INFO] WildFly Elytron - SASL Anonymous ................... SKIPPED
[INFO] WildFly Elytron - SASL Digest ...................... SKIPPED
[INFO] WildFly Elytron - SASL Entity ...................... SKIPPED
[INFO] WildFly Elytron - Tests Common ..................... SKIPPED
[INFO] WildFly Elytron - SASL External .................... SKIPPED
[INFO] WildFly Elytron - SASL GS2 ......................... SKIPPED
[INFO] WildFly Elytron - SASL GSSAPI ...................... SKIPPED
[INFO] WildFly Elytron - SASL JBOSS-LOCAL-USER ............ SKIPPED
[INFO] WildFly Elytron - Mechanism OAuth2 ................. SKIPPED
[INFO] WildFly Elytron - SASL OAuth2 ...................... SKIPPED
[INFO] WildFly Elytron - SASL OTP ......................... SKIPPED
[INFO] WildFly Elytron - SASL Plain ....................... SKIPPED
[INFO] WildFly Elytron - Mechanism SCRAM .................. SKIPPED
[INFO] WildFly Elytron - SASL SCRAM ....................... SKIPPED
[INFO] WildFly Elytron - X.500 Certificate ACME ........... SKIPPED
[INFO] WildFly Elytron - Tests ............................ SKIPPED
[INFO] WildFly Elytron - HTTP OIDC ........................ SKIPPED
[INFO] WildFly Elytron - HTTP SSO ......................... SKIPPED
[INFO] WildFly Elytron - Security Manager ................. SKIPPED
[INFO] WildFly Elytron - SASL Auth Utility Classes ........ SKIPPED
[INFO] WildFly Elytron - SASL Deprecated .................. SKIPPED
[INFO] WildFly Elytron Tool ............................... SKIPPED
[INFO] WildFly Elytron - X.500 Deprecated ................. SKIPPED
[INFO] WildFly Elytron .................................... SKIPPED
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  02:13 min
[INFO] Finished at: 2023-10-07T01:04:55+05:30
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-surefire-plugin:2.19.1:test (default-test) on project wildfly-elytron-ssl: There are test failures.
[ERROR] 
[ERROR] Please refer to /home/abhishek/Code/wildfly-elytron-2/ssl/target/surefire-reports for the individual test results.
[ERROR] -> [Help 1]
[ERROR] 
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR] 
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MojoFailureException
[ERROR] 
[ERROR] After correcting the problems, you can resume the build with the command
[ERROR]   mvn <args> -rf :wildfly-elytron-ssl
```

--- a/auth/client/src/main/java/org/wildfly/security/auth/client/ElytronXmlParser.java
+++ b/auth/client/src/main/java/org/wildfly/security/auth/client/ElytronXmlParser.java
@@ -714,13 +714,10 @@ public final class ElytronXmlParser {
         while (reader.hasNext()) {
             final int tag = reader.nextTag();
             if (tag == START_ELEMENT) {
-                switch (reader.getLocalName()) {
-                    case "certificate-revocation-list": {
-                        parseCertificateRevocationList(reader, builder, xmlVersion, true);
-                        break;
-                    }
-                    default:
-                        throw reader.unexpectedElement();
+                if (reader.getLocalName().equals("certificate-revocation-list")) {
+                    parseCertificateRevocationList(reader, builder, xmlVersion, true);
+                } else {
+                    throw reader.unexpectedElement();
                 }
             } else if (tag != END_ELEMENT) {
                 throw reader.unexpectedContent();

--- a/auth/realm/base/src/main/java/org/wildfly/security/auth/realm/FileSystemSecurityRealm.java
+++ b/auth/realm/base/src/main/java/org/wildfly/security/auth/realm/FileSystemSecurityRealm.java
@@ -1099,6 +1099,7 @@ public final class FileSystemSecurityRealm implements ModifiableSecurityRealm, C
             streamWriter.writeEndDocument();
         }
 
+        @Override
         public void dispose() {
             // Release the lock for this realm identity
             IdentityLock identityLock = lock;

--- a/tool/src/main/java/org/wildfly/security/tool/VaultCommand.java
+++ b/tool/src/main/java/org/wildfly/security/tool/VaultCommand.java
@@ -446,7 +446,7 @@ public class VaultCommand extends Command {
         if (keystorePassword != null) {
             password = keystorePassword;
             if (salt != null && iterationCount > -1) {
-                password = keystorePassword.startsWith(MASK_PREFIX) ? keystorePassword + ";" + salt + ";" + String.valueOf(iterationCount)
+                password = keystorePassword.startsWith(MASK_PREFIX) ? keystorePassword + ";" + salt + ";" + iterationCount
                         : MaskCommand.computeMasked(keystorePassword, salt, iterationCount);
             }
         }

